front/components/assistant_builder/AssistantBuilder.tsx

```diff
@@ -132,9 +176,12 @@ export default function AssistantBuilder({
       return;
     }
 
-    // In case of duplicating an agent, we should not override the original actions from
-    // the fetched actions (= empty actions).
-    if (actions.length > 0) {
+    // In case of duplicating an agent, we set initial action state from the original agent.
+    // We should not override it with the actions we fetched from the client side (= empty actions).
+    if (
+      initialBuilderState?.actions &&
+      initialBuilderState.actions.length === 0
+    ) {
```

Comment:
We only override the action state when there is no initial actions is defined (= not duplicating an agent)